### PR TITLE
Add 8bim paths event listener

### PR DESCRIPTION
**Pull request by matslindh**
This event listener extracts paths from the 8BIM metadata and adds them as metadata under the "paths" element in the metadata store.

It currently only extracts the paths, but if necessary in the future can be expanded to also extract and parse additional fields. In that cases we should consider a separate "8bim:" namespace for metadata.

### CODE REVIEW

File: src/EventListener/EightbimMetadata.php [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+namespace Imbo\EventListener;
+
+use Imbo\EventManager\EventInterface,
+    Imbo\EventListener\ListenerInterface,
+    Imbo\Exception\RuntimeException,
+    Imbo\Exception\DatabaseException;
+
+/**
+ * 8bim metadata event listener
```
Comment (christeredvartsen) [inline src/EventListener/EightbimMetadata.php]: Inconsequent use of 8bim vs 8BIM throughout the file.

File: src/EventListener/EightbimMetadata.php [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+     * Metadata extracted from 8BIM profile
+     *
+     * @var array
+     */
+    protected $properties = [];
+
+    /**
+     * Set the Imagick instance to be used - the instance will be cloned to avoid polluting the existing imagick object.
+     *
+     * @param $imagick
```
Comment (christeredvartsen) [inline src/EventListener/EightbimMetadata.php]: `@param \Imagick $imagick`

File: src/EventListener/EightbimMetadata.php [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+     * Read 8bim data from incoming image
+     *
+     * @param EventInterface $event The triggered event
+     * @return array
+     */
+    public function populate(EventInterface $event) {
+        $image = $event->getRequest()->getImage();
+
+        // This can be replaced with ImagickAware and clone $imagick when the ImagickAware patch has been merged
+        $imagick = new \Imagick();
```
Comment (christeredvartsen) [inline src/EventListener/EightbimMetadata.php]: Use `$this->imagick` instead as the `ImagickAware` interface is now available.

File: tests/behat/imbo-configs/add-8bim-data-as-metadata.php [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+ * This file is part of the Imbo package
+ *
+ * (c) Christer Edvartsen <cogo@starzinger.net>
+ *
+ * For the full copyright and license information, please view the LICENSE file that was
+ * distributed with this source code.
+ */
+
+/**
+ * Enable the EXIF metadata listener
```
Comment (christeredvartsen) [inline tests/behat/imbo-configs/add-8bim-data-as-metadata.php]: Copy / paste error

File: tests/phpunit/unit/EventListener/EightbimMetadataTest.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+ * For the full copyright and license information, please view the LICENSE file that was
+ * distributed with this source code.
+ */
+
+namespace ImboUnitTest\EventListener;
+
+use Imbo\EventListener\EightbimMetadata;
+
+/**
+ * @covers Imbo\EventListener\ExifMetadata
```
Comment (christeredvartsen) [inline tests/phpunit/unit/EventListener/EightbimMetadataTest.php]: `@covers` refers to the wrong class

File: tests/phpunit/unit/EventListener/EightbimMetadataTest.php [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getListener() {
+        return $this->listener;
+    }
+
+    public function testCanExtractMetadata() {
```
Comment (christeredvartsen) [inline tests/phpunit/unit/EventListener/EightbimMetadataTest.php]: Add `@covers` annotation(s) for the method(s) this test covers.

File: tests/phpunit/unit/EventListener/EightbimMetadataTest.php [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        $database->expects($this->once())->method('updateMetadata')->with($user, $imageIdentifier, [
+            'paths' => ['House', 'Panda'],
+        ]);
+
+        $event = $this->createMock('Imbo\EventManager\Event');
+        $event->expects($this->exactly(2))->method('getRequest')->will($this->returnValue($request));
+        $event->expects($this->once())->method('getDatabase')->will($this->returnValue($database));
+
+        $listener = new EightbimMetadata();
+        $listener->populate($event);
```
Comment (christeredvartsen) [inline tests/phpunit/unit/EventListener/EightbimMetadataTest.php]: Add assertion to the return value as this is documented above to be an array?

**Comment by matslindh** [comment]
All requested changes have been pushed! 👍

**Pull request merged**